packages/react-component-library/src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+
+interface DrawerProps extends ComponentWithClass {
+  children?: React.ReactNode
+  onClose?: (event: React.FormEvent<HTMLButtonElement>) => void
+  isOpen?: boolean
+}
+
+export const Drawer: React.FC<DrawerProps> = ({
+  children,
+  onClose,
+  isOpen = false,
+}) => {
+  const [drawerOpen, setDrawerOpen] = useState(isOpen)
+
+  const classes = classNames('rn-drawer', {
+    'is-open': drawerOpen,
+  })
+
+  function handleClick(event: React.FormEvent<HTMLButtonElement>) {
+    setDrawerOpen(false)
+    onClose(event)
+  }
+
+  return (
+    <div className={classes} data-testid="drawer-wrapper">
+      <div className="rn-drawer__inner">
+        <button
+          className="rn-drawer__close"
+          onClick={handleClick}
+          data-testid="drawer-close"
```

Comment:
I wonder if this button can be found by using the text rather than `data-testid`..? I feel that `data-testid` should be a last resort

Reply:
It's slightly annoying to target because i'm using the HTML character code `&times;`.

I figure by using the testid we can modify the implementation without breaking the test, so they are resilient to change?